modules/dxp/apps/analytics/analytics-reports-layout-impl/src/main/java/com/liferay/analytics/reports/layout/internal/info/item/LayoutAnalyticsReportsInfoItem.java

```diff
@@ -80,27 +82,31 @@ public long getAuthorUserId(Layout layout) {
 
 	@Override
 	public WebImage getAuthorWebImage(Layout layout, Locale locale) {
-		ThemeDisplay themeDisplay = _getThemeDisplay();
-
-		if (themeDisplay == null) {
-			return new WebImage(StringPool.BLANK);
-		}
-
-		Optional<User> userOptional = _getUserOptional(layout);
-
-		return userOptional.map(
-			user -> {
-				try {
-					return new WebImage(user.getPortraitURL(themeDisplay));
-				}
-				catch (PortalException portalException) {
-					_log.error(portalException, portalException);
-
-					return new WebImage(StringPool.BLANK);
-				}
+		Optional<ThemeDisplay> themeDisplayOptional =
+			_getThemeDisplayOptional();
+
+		return themeDisplayOptional.map(
+			themeDisplay -> {
+				Optional<User> userOptional = _getUserOptional(layout);
+
+				return userOptional.map(
+					user -> {
+						try {
+							return new WebImage(
+								user.getPortraitURL(themeDisplay));
+						}
+						catch (PortalException portalException) {
+							_log.error(portalException, portalException);
+
+							return new WebImage(StringPool.BLANK);
+						}
+					}
+				).orElse(
```

Comment:
Should it be used `orElseGet` too for consistency?

Reply:
orElseGet should never be used with the excuse of consistency, this is a very dangerous comment. The reason to use orElseGet or orElse it is related with the fact that the orElse will be always called, and the orElseGet method is called only when the result is null.

In this case, make sense to use the orElseGet too.

But you should be very carefull to use consistency in order to decide what is the best tool for the job.

Reply:
I agree, we have had bad moments with that, I just said the consistency reason because the creation of an empty WebImage is not expensive at `all`

Reply:
We don't know the cost of the creation of a WebImage, have we run performance test for this issue?

Reply:
AFAIK we have not. I got your idea, I shouldn't take it for granted ;)